packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton/index.tsx

```diff
@@ -21,6 +24,14 @@ export const RainbowKitCustomConnectButton = () => {
   const { address: connectedAddress } = useAccount();
   const { data: user, isLoading: isLoadingUser } = useUser(connectedAddress);
   const { disconnect } = useDisconnect();
+  const router = useRouter();
+  const { isAuthenticated } = useAuthSession();
+
+  useEffect(() => {
```

Comment:
This causes infinite reloadings for notFound pages, see example with `/asdf`

https://github.com/user-attachments/assets/14dc3def-81fd-4d43-afb4-e3a1ddaa0585

Reply:
https://github.com/user-attachments/assets/5b5ec66d-e7b3-43f1-9772-42b5f5b2bb4c

For me it's not happening, like I see you disconnected from wallet. Ideally if you disconnect the `isAuthenticated` will be false only right? Like even if you disconnect from wallet UI 

The logic from https://github.com/BuidlGuidl/ens-pg/pull/55 will remove the session and `isAuthenticated` will be false for sure 🙌

Reply:
My bad it reloads when connected as admin. Connect button doesn't show it since it's always loading after page reload

Reply:
rechecked, works as expected!

Reply:
Ohhhhh I see just fixed https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/159/commits/9e14f13abd5f80c302b7e7f18cbb8b0b0a56b417 


And since we know that admin will authenticate on `/admin/*` routes only. Since we have `router.refresh()` in `SingIn` component it will fetch the admin data. 

--- 

The problem with Not Found page is that it doesn't seem to remember the session when we do `route.refresh()`. While other pages when `isAuthenticated` becomes true and we do `router.refresh()` the `isAuthenticated` remains true (and since useEffect dependency does not change we don't get the infinite loop)